### PR TITLE
PKG-4701 initial v12.4

### PR DESCRIPTION
**Pull request by danpetry**
**Destination channel:** defaults

### Links

- [PKG-4701](https://anaconda.atlassian.net/browse/PKG-4701)

### Explanation of changes:

- Part of CUDA 12.4 build, following [conda-forge's pattern](https://github.com/conda-forge/staged-recipes/issues/21382) which was developed in conjunction with Nvidia. I don't see a need to stage these then release them all at once. They won't cause a problem released one at a time.
- Please check the main branch too, this has just been forked
- Changes made to adjust feedstocks for differences between conda-forge and defaults. Please see commit messages for description of changes.

[PKG-4701]: https://anaconda.atlassian.net/browse/PKG-4701?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cbouss — COMMENTED** [review]
Will it work without an ocl loader?
I see this one too: https://github.com/conda-forge/ocl-icd-system-feedstock/tree/main
Which seems to indicate having a system ocl would work out of the box.
Is that package (cuda-opencl) needed for other tasks?

**Comment by danpetry** [comment]
The loader's a broker between different OpenCL implementations and your application. It's needed if there are multiple opencl implementations on the system. Which is a common occurrence AFAIU. It's possible to build applications that link directly to the OpenCL implementation but then it will only work with that one which seems to defeat the purpose (it's supposed to allow portability between different hardware architectures). Having said that, cuda-opencl is called by the ICD loader, not the other way around. So I'm not sure if it's accurate to express the ICD loader as a dependency of cuda-opencl, rather should be up to the user to manage this aspect. Even though in practical scenarios it's needed.

**Comment by danpetry** [comment]
This package isn't used anywhere else apart from being grouped in the `cuda-libraries` metapackage.

I can see why conda-forge added the loader. I'm open as to whether we should. It doesn't look too difficult.

**Comment by JeanChristopheMorinPerso** [comment]
@danpetry DO you have a link to some documentation that could help us understand what's at play? I'm not verry familiar with all of this (read I have zero knowledge).

**Comment by danpetry** [comment]
https://www.khronos.org/news/permalink/opencl-installable-client-driver-icd-loader

**Comment by danpetry** [comment]
I'll address the dependencies and come back to this discussion if it's looking like a headache for some reason

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ source:
 
 build:
   number: 0
-  skip: true  # [osx or aarch64 or ppc64le]
+  skip: true  # [osx or (linux and s390x) or aarch64 or ppc64le]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
  skip: true  # [osx or (linux and (s390x or aarch64 or ppc64le))]
```

**Review by psteyer — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**